apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/AbstractCleanupJob.java

```diff
@@ -0,0 +1,162 @@
+package gov.cms.bfd.pipeline.rda.grpc;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Query;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.slf4j.Logger;
+
+/**
+ * An abstract class that encapsulates the common code for executing a job to clean up old
+ * pre-adjudicated claims from the RDA pipeline tables.
+ */
+@Getter
+@AllArgsConstructor
+public abstract class AbstractCleanupJob implements CleanupJob {
+
+  /** maximum age of claims in days from current date. */
+  private static final int OLDEST_CLAIM_AGE_IN_DAYS = 60;
+
+  /** template for delete query. */
+  private static final String DELETE_QUERY_TEMPLATE =
+      "delete from %s t where t.%s in ("
+          + "  select %s "
+          + "  from %s "
+          + "  where last_updated < '%s' "
+          + "  order by last_updated desc "
+          + "  limit %d"
+          + ")";
+
+  /** EntityManagerFactory to use to generate the EntityManager. */
+  private final EntityManagerFactory entityManagerFactory;
```

Comment:
It would be better to use a `TransactionManager` rather than deal with transactions, aborts, etc in this class.

Reply:
Done.